powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OperationServiceBehavior.java

```diff
@@ -791,6 +803,12 @@ private static boolean proximityCheckPassed(final ProximityCheckResult proximity
         return proximityCheckResult == ProximityCheckResult.SUCCESS || proximityCheckResult == ProximityCheckResult.DISABLED;
     }
 
+    private static boolean activationIdMatches(final OperationApproveRequest operationApproveRequest, String activationId) {
+        return activationId == null ||
+                (operationApproveRequest.getAdditionalData().get("activationId") != null &&
+                        operationApproveRequest.getAdditionalData().get("activationId").equals(activationId));
```

Comment:
🤔 Couldn't it be simplified?

```suggestion
        return activationId == null || activationId.equals(operationApproveRequest.getAdditionalData().get("activationId"));
```